internal/apis/certmanager/types_issuer.go

```diff
@@ -197,6 +197,16 @@ type VaultIssuer struct {
 	// If no key for the Secret is specified, cert-manager will default to 'ca.crt'.
 	// +optional
 	CABundleSecretRef *cmmeta.SecretKeySelector
+
+	// Reference to a Secret containing a PEM-encoded Client Certificate to use when the
+	// Vault server requires mTLS.
+	// +optional
+	ClientCertSecretRef *cmmeta.SecretKeySelector
```

Comment:
I **think** this should say "PEM-encoded Client Certificate **chain**" because the client needs to present the client certificate and any intermediate certificates that complete the chain to the CA certificate that the Vault server is using to verify the client cert.

> // Certificates contains one or more certificate chains to present to the
> // other side of the connection. The first certificate compatible with the
> // peer's requirements is selected automatically.

https://pkg.go.dev/crypto/tls#Config

Reply:
I also wonder if if should say (and support) "chain**s**" (plural).
Consider the case where cert-manager is connecting to a  Vault **service** backed by a cluster of Vault servers.
If a new client CA certificate has been created and is being rolled out, the vault servers may be using a mixture of original and new client CA certificates, so the clients would each need client certificates chains for both the original and the new client CAs....at least until all the servers were using the new CA.
I don't know if that's a realistic scenario. Please explain briefly how you intend to rotate the client CA in your infrastructure.

Reply:
Hi @wallrj , thanks for your comment, while what you said that a chain of multiple client certificates could be provided, the client certificate don't have to present the CA certificate once again, client certificates only contains the certificate of the client itself, the CA certificate is configured at server side under the `tls_client_ca_file` configuration of the TCP listener as documented [here](https://developer.hashicorp.com/vault/docs/configuration/listener/tcp#tls_client_ca_file), and the server is responsible for validating if any of the client certificates provided matches the CA configured at server side.

Circling back to your suggestion of adding the word "chain" to the description, I can definitely do that, but realistic, based on what I said above, a client certificate is composed by only one certificate.

> Please explain briefly how you intend to rotate the client CA in your infrastructure.
- The initial work is done at Server side, you have to update the CA certificate bundle file used in the `tls_client_ca_file` configuration to include the new CA certificate chain as following:
  - you will keep the existing CA bundle in the file for existing client certificates to keep working
  - you will add the new CA bundle to get the server start trusting the new client certificates issued by it.
- Once you have the server updated with the new client CA bundle, you can start updating the client related secrets and adding the new client certificate and private keys, but once again, there is no need for providing the client CA bundled in the client certificate, the server already have it and will be responsible for validating the client certificate.